src/line.cpp

```diff
@@ -289,7 +289,8 @@ direction direction_from(int const x1, int const y1, int const x2, int const y2)
 
 direction direction_from(tripoint const &p, tripoint const &q)
 {
-    return direction_from(q.x - p.x, q.y - p.y, q.z - p.z);
+    // Note: Z coord has to be inverted either here or in direction defintions
```

Comment:
I'd argue that this fix should go into the direction enum directly. The enum establishes a connection between the sign of the numbers and its observed meaning - a z-offset of -1 means down, just like a x-offset of -1 means west (left).

But this would change the enumerations numeric values, which is known to cause trouble (the older ones might remember the confused monster compass), so the fix here is probably good for now.